### DISCUSSION TITLE
AMR and Presplit Mesh

### DISCUSSION THREAD

**Discussion by makeclean** [idaholab/moose#17367] [Q&A General] [answered]
Maybe I'm missing something obvious, but why cant one use Adaptivity & Distributed mesh?

```
*** ERROR ***
The following error occurred in the object "mesh", of type "FileMesh".

Cannot use StatefulMaterials + Adaptivity with DistributedMesh!
Consider specifying parallel_type = 'replicated' in your input file
to prevent it from being run with DistributedMesh.
```
Large problems basically require pre-splitting and then use of the split mesh, right?

**Comment by GiudGiud** [comment, marked as answer]
Large problems will indeed require distributed meshes.
Looks related to https://github.com/idaholab/moose/issues/4532

@permcody or @friedmud will know more.

**Reply by GiudGiud** [reply]
As I have been using this system more lately, pre-splitting is only useful if the mesh is so big that it can't fit on a single node. This is only for very large problems.
It's also a good optimization to reduce IO on slower filesystems.

**Comment by makeclean** [comment]
Yes, AMR would be very useful in these situations too though right?
________________________________

**Reply by GiudGiud** [reply]
Well I imagine there's a good reason for your use of stateful materials so that can't change. 
If I were to choose one of the 3 to remove to make it work, it would be AMR. The mesh can be pre-refined. 
Combining the 3 is currently not supported and no one is currently working on allowing this. If you have someone work on this, we will be glad to help them enable it.
